### DISCUSSION TITLE
Imputing scATAC data into spatial transcriptomic data - issue with the fragment file

### DISCUSSION THREAD

**Discussion by a-schot** [stuart-lab/signac#725] [Q&A] [answered]
Hi @timoast,

I have been experimenting with imputing scATAC data into spatial transcription data. For testing this, I used the SNARE-seq vignette data and was successful in imputing the peak matrix following the [multimodal reference mapping vignette](https://satijalab.org/seurat/articles/multimodal_reference_mapping.html) in Seurat (see code below). The thing is, the ChromatinAssay object relies on the fragment file, but the fragment file isn't imputed, just the peak matrix is. Using the original fragment file isn't an option as the fragments are specific to the cell names in the reference dataset.

Do you know of a solution for this problem? I would like to be able to create a ChromatinAssay object from the imputed scATAC data if at all possible.

Thank you!

```r
cortex <- MapQuery(
  anchorset = anchors,
  query = cortex,
  reference = snare,
  refdata = list(celltype = "celltype",
                 predicted_ATAC = "ATAC"),
  reference.reduction = "spca", 
  reduction.model = "wnn.umap",
)
```

**Comment by timoast** [comment, marked as answer]
Using this method, you're able to impute peak accessibility in the spatially-resolved cells. There's no way to really impute the whole fragment file, since this is genome-wide base-resolution data for each cell.

If your goal is just to create a `ChromatinAssay` the fragment file is not required for that (ie, you can create a `ChromatinAssay` with no fragment file)

**Reply by a-schot** [reply]
Thanks for the response!

I've tested creating a ChromatinAssay object with the peak data alone, but ran into an error. I guess I kind of assumed it was due to the fragment file not being present, good to know that isn't the case. Could you maybe shed some insight into what is causing the issue here? 


### code
```r
predicted_ATAC_chromassay <- CreateChromatinAssay(
  data = cortex[["predicted_ATAC"]],
)
```
### output
```
Error in h(simpleError(msg, call)) : 
  error in evaluating the argument 'x' in selecting a method for function 'colSums': comparison (6) is possible only for atomic and list types
```

**Reply by a-schot** [reply]
Found the answer! The issue was that CreateChromatinAssay() doesn't work with Assay objects as input, and that's the class the ATAC data has after being imputed. I just had to pull out the "data" matrix by using `cortex@assays$predicted_ATAC@data` to pass to `CreateChromatinAssay()` it worked!
